ironfish/src/rpc/clients/socketClient.ts

```diff
@@ -286,6 +286,7 @@ export abstract class RpcSocketClient extends RpcClient {
     this.isConnected = true
     this.client.on('data', this.onClientData)
     this.client.on('close', this.onClientClose)
+    this.client.on('error', this.onError.bind(this))
```

Comment:
Do you want to `.off` this in `onClientClose` too? I don't think it's actually necessary, but we do it for the `data` and `close` events.